languagetool-server/src/main/java/org/languagetool/server/TextChecker.java

```diff
@@ -572,6 +577,32 @@ void checkText(AnnotatedText aText, HttpExchange httpExchange, Map<String, Strin
       hiddenMatches.addAll(ResultExtender.getAsHiddenMatches(allMatches, premiumMatches));
     }
 
+    //### Start multiLangPart
+    //TODO: implement recheck of ignoreRanges
+    if (isMultiLangEnabled) {
+      long startTimeRecheck = System.currentTimeMillis();
+      Map<String, List<Range>> rangesOrderedByLanguage = new HashMap<>();
+      res.forEach(checkResults -> {
+        checkResults.getIgnoredRanges().forEach(range -> {
+          List<Range> sentenceRanges = rangesOrderedByLanguage.getOrDefault(range.getLang(), new ArrayList<>());
+          sentenceRanges.add(range);
+          rangesOrderedByLanguage.put(range.getLang(), sentenceRanges);
+        });
+      });
+      rangesOrderedByLanguage.forEach((shortLangCode, ranges) -> {
```

Comment:
I don't think appending all ranges together here is correct when checking the text, since these could be distinct sentences separated by text in other languages; this might lead to e.g. false positives from text-level rules.

Reply:
You are right. As this should not run (the parameter MultiLangEnabled is not used now), I will remove this part.